src/Console/Command/AbstractCommand.php

```diff
@@ -3,12 +3,72 @@
 
 namespace duckpony\Console\Command;
 
+use Monolog\Logger;
+use Psr\Log\LoggerInterface;
 use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Yaml\Yaml;
+use Throwable;
 
-class AbstractCommand extends Command
+abstract class AbstractCommand extends Command
 {
+    use SlackLoggerAwareTrait;
+
+    /** @var Logger */
+    private $logger;
+
+    public function __construct(Logger $logger)
+    {
+        parent::__construct();
+        $this->logger = $logger;
+    }
+
+    protected function configure()
+    {
+        parent::configure();
+        $this->addOption('config',
+            'c',
+            InputOption::VALUE_REQUIRED,
+            'Config',
+            $this->getRootPath() . '/config/config.yml');
+    }
+
     protected function getRootPath(): string
     {
         return stream_resolve_include_path(__DIR__ . '/../../../');
     }
+
+    final protected function execute(InputInterface $input, OutputInterface $output)
+    {
+        $config = Yaml::parse(file_get_contents($input->getOption('config')));
+        $this->initSlackLogger($this->logger, $config);
+
+        try {
+            return $this->executeWithConfig($input, $output, $this->logger, $config);
```

Comment:
Why pass the logger into this method? Would you instead simply add a `getLogger` method, or simply mage the logger `protected` to access it from derived classes.

Reply:
Because it gives you a better overview of the dependencies needed. I would prefer it this way to decouple the dependencies between the sub and parent class, but a better way would be to decouple the tasks from the command altogether and create services using DI. See #12 where I started this and #13

Reply:
> Because it gives you a better overview of the dependencies needed

This would be true if you were using this in different service/class calls.
But since the derived class always gets the logger assigned, there is simply not need to pass this into an abstract method of the same type.

Reply:
But the goal should be to exract the code from the Command handler, and then we will have to pass the logger - this was the way I was thinking when I did it this way.

Reply:
I don't think that you will pass the logger right into the function, but rather pass the logger into the constructor using DI. So there is not need for it.